include/zenoh-pico/system/link/raweth.h

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright (c) 2022 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+#ifndef ZENOH_PICO_SYSTEM_LINK_RAWETH_H
+#define ZENOH_PICO_SYSTEM_LINK_RAWETH_H
+
+#include <stdint.h>
+
+#include "zenoh-pico/collections/string.h"
+#include "zenoh-pico/system/platform.h"
+
+#if Z_FEATURE_RAWETH_TRANSPORT == 1
+
+// Ethernet types
+#define _ZP_ETH_TYPE_VLAN 0x8100   // 2048  (0x0800) IPv4
+#define _ZP_ETH_TYPE_CUTOFF 0x600  // Values above are ethertype
+
+// Address Sizes
+#define _ZP_MAC_ADDR_LENGTH 6
+
+// Max frame size
+#define _ZP_MAX_ETH_FRAME_SIZE 1500
+
+typedef struct {
+    uint16_t tpid;      // Tag ethertype
+    uint16_t pcp : 3;   // Priority code point
```

Comment:
I wonder about the portability of this code. As far as I know, bitfields are not guaranteed to be portable (see [here](https://stackoverflow.com/questions/1490092/c-c-force-bit-field-order-and-alignment)) between different platforms, endianness, and compilers. Should we just use `uint16_t` and do some bitmasking?

Reply:
Yeah portability isn't guaranteed in bitfields, now since we don't want to send this structure, the problem mainly resides in platform-dependent behaviors and bugs.

Bitmasking would be more portable but introduce more clutter and possibility for human mistakes.

Reply:
Now the other thing is, we don't really need to interact with the vlan tag we just need to write it as specified in a config (or locator in the future) so we might as well remove this structure entirely.

Reply:
If it is not needed, then I suggest to remove it.